solr/test-framework/src/java/org/apache/solr/SolrTestCaseJ4.java

```diff
@@ -2147,34 +2247,51 @@ public static void assertXmlFile(final File file, String... xpath)
       String xml = Files.readString(file.toPath());
       String results = TestHarness.validateXPath(xml, xpath);
       if (null != results) {
-        String msg = "File XPath failure: file=" + file.getPath() + " xpath="
-            + results + "\n\nxml was: " + xml;
+        String msg =
```

Comment:
minor: I suspect inlining msg to the `fail(....)` would be _just_ short enough to not overflow

Reply:
Checked and its still too long.